hawk/app/models/report.rb

```diff
@@ -1,69 +1,225 @@
 # Copyright (c) 2009-2015 Tim Serong <tserong@suse.com>
 # See COPYING for license.
 
+require 'fileutils'
+
 class Report
   attr_accessor :id
-  attr_accessor :order
-
+  attr_accessor :name
   attr_accessor :archive
-  attr_accessor :from
-  attr_accessor :until
+  attr_accessor :from_time
+  attr_accessor :to_time
 
   def initialize(attributes)
-    attributes.each do |key, value|
-      self.send(
-        "#{key}=".to_sym,
-        value
-      )
-    end
+    attributes.each { |key, value| send("#{key}=".to_sym, value) }
   end
 
-  def delete
+  def delete(hb_report)
     self.archive.delete
+    FileUtils.remove_entry_secure(hb_report.path) if File.exists?(hb_report.path)
+    FileUtils.remove_entry_secure(hb_report.outfile) if File.exists?(hb_report.outfile)
+    FileUtils.remove_entry_secure(hb_report.errfile) if File.exists?(hb_report.errfile)
   end
 
   def load!
+  end
+
+  def report_path
+    self.class.report_path
+  end
+
+  def mimetype
+    if archive.extname == ".bz2"
+      "application/x-bzip"
+    elsif archive.extname == ".xz"
+      "application/x-xz"
+    elsif archive.extname == ".gz"
+      "application/x-gz"
+    else
+      "application/x-compressed"
+    end
+  end
+
+  def transitions(hb_report)
```

Comment:
Metrics/AbcSize: Assignment Branch Condition size for transitions is too high. [35.62/15]